src/test/ui/suggestions/format-borrow.stderr

```diff
@@ -12,11 +12,17 @@ error[E0308]: mismatched types
   --> $DIR/format-borrow.rs:4:21
    |
 LL |     let b: String = &format!("b");
-   |            ------   ^^^^^^^^^^^^^
-   |            |        |
-   |            |        expected struct `std::string::String`, found `&std::string::String`
-   |            |        help: consider removing the borrow: `format!("b")`
+   |            ------   ^^^^^^^^^^^^^ expected struct `std::string::String`, found `&std::string::String`
+   |            |
    |            expected due to this
+   |
+help: consider removing the borrow
+   |
+LL |     let b: String = {
+LL |         let res = $crate::fmt::format($crate::__export::format_args!($($arg)*));
+LL |         res
+LL |     };
```

Comment:
Hmm can we avoid this in suggestions? It's definitely doing the wrong thing, looks like it's mixing source code from two spans?